src/auth/progress.auth.sso.js

```diff
@@ -242,17 +261,64 @@ limitations under the License.
 
 
         // Override the protoype's method but call it from within the override. (Define the override 
-        // here in the constructor so it has access to the internal function)
-        this._openRequestAndAuthorize =
-            function (xhr, verb, uri) {
-            
-                progress.data.AuthenticationProviderSSO.prototype._openRequestAndAuthorize.apply(
-                    this,
-                    [xhr, verb, uri]
-                );
-            
-                xhr.setRequestHeader('Authorization', "oecp " + getToken());
-            };
+        // here in the constructor so it has access to the internal function getToken() )
+        this._openRequestAndAuthorize = function (xhr,
+                                                  verb,
+                                                  uri,
+                                                  callback) {
+            var that = this,
+                date,
+                errorObject;
+
+            function afterRefreshCheck(provider, result, info) {
```

Comment:
I don't like the afterRefreshCheck() name. It doesn't really accurately describe what it does, especially since this is what actually does the logic that afterRefreshCheck(). Would it make sense to separate the refresh check from the actual openRequestandAuthorize() call?